### PR TITLE
Fix: Update packaging

### PR DESCRIPTION
**Pull request by oxedions**
Update the whole packaging system to be more efficient.

### CODE REVIEW

File: packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe [hunk 1]
```diff
@@ -1,30 +1,9 @@
...
-echo . . $R@i.~~.!. . .:. .~$$$$$B$$en:``. . . . . . . . ..
-echo . . ?MXT@Wx.~. . :. . .~"##*$$$$M~. . . . . . . . . ..
+
 echo
-echo +----------------------------------------------------+
-echo | . . . . . . . iPXE embed - dhcpretry . . . . . . . |
-echo | . . . . . BlueBanquise - Benoit Leveugle . . . . . |
-echo +----------------------------------------------------+
+echo . . . +----------------------------------------------------+ . . .
+echo . . . | . . . . . . . iPXE embed - standard. . . . . . . . | . . .
```
Comment (btravouillon) [inline packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe:4]: Change from dhcpretry to standard in file bluebanquise_dhcpretry.ipxe: is it expected?

File: packages/karma/karma.spec [hunk 2]
```diff
@@ -1,10 +1,10 @@
 Name:     karma
 Summary:  karma
 Release:  1%{?dist}
-Version:  0.68
+Version:  %{_software_version}
 License:  apache-2.0
 Group:    System Environment/Base
-Source:   https://github.com/prymitive/karma/releases/download/v0.68/karma-linux-amd64.tar.gz
+Source:   https://github.com/prymitive/karma/releases/download/v%{_software_version}/karma.tar.gz
```
Comment (btravouillon) [inline packages/karma/karma.spec:7]: The name of the source file mismatch the name or the wget target below.

File: packages/prometheus/prometheus.spec [hunk 3]
```diff
@@ -0,0 +1,59 @@
+Name:     prometheus
```
Comment (btravouillon) [inline packages/prometheus/prometheus.spec:1]: This is a new file.
Below I can see: packages/prometheus-2.20.1/prometheus.spec → packages/prometheus/prometheus.spec.bkp 

I guess you don't want to commit the backup file to the git project.

File: parameters.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Number of cores
```
Comment (btravouillon) [inline parameters.sh:1]: I would rename the file `parameters` or `parameters.conf`

File: versions.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+ipxe_bluebanquise_version=1.1.3
```
Comment (btravouillon) [inline versions.sh:1]: I would rename the file `versions` or `versions.conf`

File: packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe (hunk 1 above)
Comment (oxedions) [reply packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe:4]: Good catch. Solved.

File: packages/karma/karma.spec (hunk 2 above)
Comment (oxedions) [reply packages/karma/karma.spec:7]: Solved

File: packages/prometheus/prometheus.spec (hunk 3 above)
Comment (oxedions) [reply packages/prometheus/prometheus.spec:1]: indeed. removed.

File: parameters.sh (hunk 4 above)
Comment (oxedions) [reply parameters.sh:1]: done

File: versions.sh (hunk 5 above)
Comment (oxedions) [reply versions.sh:1]: done

File: packages/ansible-cmdb/ansible-cmdb.spec [hunk 6]
```diff
@@ -1,10 +1,10 @@
 Name:     ansible-cmdb
 Summary:  ansible-cmdb
 Release:  1%{?dist}
-Version:  1.30
+Version:  %{_software_version}
 License:  GPLv3
 Group:    System Environment/Base
-Source:   https://github.com/fboender/ansible-cmdb/releases/download/1.30/ansible-cmdb-1.30.tar.gz
+Source:   https://github.com/fboender/ansible-cmdb/releases/download/%{_software_version}/ansible-cmdb.tar.gz
```
Comment (btravouillon) [inline packages/ansible-cmdb/ansible-cmdb.spec:7]: 404 not found

File: packages/ipmi_exporter/ipmi_exporter.spec [hunk 7]
```diff
@@ -1,10 +1,10 @@
 Name:     ipmi_exporter
 Summary:  ipmi_exporter
 Release:  1%{?dist}
-Version:  1.3.0
+Version:  %{_software_version}
 License:  apache-2.0
 Group:    System Environment/Base
-Source:   https://github.com/soundcloud/ipmi_exporter/releases/download/v1.3.0/ipmi_exporter-v1.3.0.linux-amd64.tar.gz
+Source:   https://github.com/soundcloud/ipmi_exporter/releases/download/v%{_software_version}/ipmi_exporter.tar.gz
```
Comment (btravouillon) [inline packages/ipmi_exporter/ipmi_exporter.spec:7]: 404 not found

File: packages/node_exporter/node_exporter.spec [hunk 8]
```diff
@@ -1,10 +1,10 @@
 Name:     node_exporter
 Summary:  node_exporter
 Release:  1%{?dist}
-Version:  1.0.1
+Version:  %{_software_version}
 License:  apache-2.0
 Group:    System Environment/Base
-Source:   https://github.com/prometheus/node_exporter/releases/download/v1.0.1/node_exporter-1.0.1.tar.gz
+Source:   https://github.com/prometheus/node_exporter/releases/download/v%{_software_version}/node_exporter.tar.gz
```
Comment (btravouillon) [inline packages/node_exporter/node_exporter.spec:7]: 404 not found

File: packages/prometheus/prometheus.spec [hunk 9]
```diff
@@ -1,10 +1,10 @@
 Name:     prometheus
 Summary:  prometheus
 Release:  1%{?dist}
-Version:  2.20.1
+Version:  %{_software_version}
 License:  apache-2.0
 Group:    System Environment/Base
-Source:   https://github.com/prometheus/prometheus/releases/download/v2.20.1/prometheus-2.19.2.tar.gz
+Source:   https://github.com/prometheus/prometheus/releases/download/v%{_software_version}/prometheus.tar.gz
```
Comment (btravouillon) [inline packages/prometheus/prometheus.spec:7]: 404 not found

File: packages/snmp_exporter/snmp_exporter.spec [hunk 10]
```diff
@@ -1,10 +1,10 @@
 Name:     snmp_exporter
 Summary:  snmp_exporter
 Release:  1%{?dist}
-Version:  0.18.0
+Version:  %{_software_version}
 License:  apache-2.0
 Group:    System Environment/Base
-Source:   https://github.com/prometheus/snmp_exporter/releases/download/v0.18.0/snmp_exporter-0.18.0.linux-amd64.tar.gz
+Source:   https://github.com/prometheus/snmp_exporter/releases/download/v%{_software_version}/snmp_exporter.tar.gz
```
Comment (btravouillon) [inline packages/snmp_exporter/snmp_exporter.spec:7]: 404 not found

File: versions.conf [hunk 11]
```diff
@@ -0,0 +1,15 @@
+ipxe_bluebanquise_version=1.1.3
+ipxe_bluebanquise_release=_bull
```
Comment (btravouillon) [inline versions.conf:2]: ??

File: auto_builder.sh [hunk 12]
```diff
@@ -0,0 +1,769 @@
...
+while [ true ]
+do
+
+echo
+echo " Select operation:"
+echo
+echo " 0 - Install needed requirements to build packages"
+echo " Build packages:"
+echo "     1 - Nyancat"
+echo " 2 - Exit."
```
Comment (btravouillon) [inline auto_builder.sh:56]: 2 - Prometheus
3 - ansible-cmdb
etc ?

File: auto_builder.sh [hunk 13]
```diff
@@ -0,0 +1,769 @@
...
+
+cd $working_directory/build/ipxe/
+mkdir ipxe-arm64-bluebanquise-$ipxe_bluebanquise_version
+cp $working_directory/sources/bluebanquise/packages/ipxe-bluebanquise/ipxe-arm64-bluebanquise.spec ipxe-arm64-bluebanquise-$ipxe_bluebanquise_version/
+sed -i "s|Version:\ \ XXX|Version:\ \ $ipxe_bluebanquise_version|g" ipxe-arm64-bluebanquise-$ipxe_bluebanquise_version/ipxe-arm64-bluebanquise.spec
+sed -i "s|working_directory=XXX|working_directory=$working_directory|g" ipxe-arm64-bluebanquise-$ipxe_bluebanquise_version/ipxe-arm64-bluebanquise.spec
+tar cvzf ipxe-arm64-bluebanquise.tar.gz ipxe-arm64-bluebanquise-$ipxe_bluebanquise_version
+rpmbuild -ta ipxe-arm64-bluebanquise.tar.gz --target=noarch
+
+fi
```
Comment (btravouillon) [inline auto_builder.sh:769]: Dead code after exit. Can you delete it?

**Review by btravouillon — CHANGES_REQUESTED** [review]

File: auto_builder.sh [hunk 14]
```diff
@@ -362,7 +369,11 @@ case $value in
         #sed -i "s|Version:\ \ XXX|Version:\ \ $ipxe_bluebanquise_version|g" ipxe-x86_64-bluebanquise-$ipxe_bluebanquise_version/ipxe-x86_64-bluebanquise.spec
         sed -i "s|working_directory=XXX|working_directory=$working_directory|g" ipxe-x86_64-bluebanquise-$ipxe_bluebanquise_version/ipxe-x86_64-bluebanquise.spec
         tar cvzf ipxe-x86_64-bluebanquise.tar.gz ipxe-x86_64-bluebanquise-$ipxe_bluebanquise_version
-        rpmbuild -ta ipxe-x86_64-bluebanquise.tar.gz --target=noarch --define "_software_version $ipxe_bluebanquise_version" --define "_software_release $ipxe_bluebanquise_release"
+#        if [ -z "$ipxe_bluebanquise_version" ]
+#          rpmbuild -ta ipxe-x86_64-bluebanquise.tar.gz --target=noarch --define "_software_version 1" --define "_software_release $ipxe_bluebanquise_release"
+#        else
+          rpmbuild -ta ipxe-x86_64-bluebanquise.tar.gz --target=noarch --define "_software_version $ipxe_bluebanquise_version" --define "_software_release 1$ipxe_bluebanquise_release"
+#        fi
```
Comment (btravouillon) [inline auto_builder.sh:376]: What is the purpose of this block? Most of the code is commented.

File: packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe [hunk 15]
```diff
@@ -33,6 +33,16 @@ echo
 echo (waiting 5s and chain...)
 sleep 5
 
-chain http://${next-server}/preboot_execution_environment/convergence.ipxe || shell
+echo Trying to chain... 1/3
```
Comment (btravouillon) [inline packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe:36]: This change is not related to the update of the packaging process.

Comment (oxedions) [reply packages/ipxe-bluebanquise/bluebanquise_dhcpretry.ipxe:36]: I Know 🙁 
Simply missing time, so I did it here...
